### PR TITLE
[Merged by Bors] - feat: `insert` and `erase` lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18729

* [`order.heyting.basic`@`4e42a9d0a79d151ee359c270e498b1a00cc6fa4e`..`9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4`](https://leanprover-community.github.io/mathlib-port-status/file/order/heyting/basic?range=4e42a9d0a79d151ee359c270e498b1a00cc6fa4e..9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4)
* [`order.boolean_algebra`@`bc7d81beddb3d6c66f71449c5bc76c38cb77cf9e`..`9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4`](https://leanprover-community.github.io/mathlib-port-status/file/order/boolean_algebra?range=bc7d81beddb3d6c66f71449c5bc76c38cb77cf9e..9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4)
* [`data.set.basic`@`29cb56a7b35f72758b05a30490e1f10bd62c35c1`..`9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4`](https://leanprover-community.github.io/mathlib-port-status/file/data/set/basic?range=29cb56a7b35f72758b05a30490e1f10bd62c35c1..9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4)
* [`data.finset.basic`@`68cc421841f2ebb8ad2b5a35a853895feb4b850a`..`9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4`](https://leanprover-community.github.io/mathlib-port-status/file/data/finset/basic?range=68cc421841f2ebb8ad2b5a35a853895feb4b850a..9ac7c0c8c4d7a535ec3e5b34b8859aab9233b2f4)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Basic.lean [hunk 1]
```diff
@@ -2200,20 +2216,96 @@ theorem union_sdiff_self (s t : Finset α) : (s ∪ t) \ t = s \ t :=
   sup_sdiff_right_self
 #align finset.union_sdiff_self Finset.union_sdiff_self
 
+-- TODO: Do we want to delete this lemma and `finset.disj_union_singleton`,
+-- or instead add `finset.union_singleton`/`finset.singleton_union`?
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/Basic.lean:2220]: These names need mathlib4-ifying

File: Mathlib/Data/Finset/Basic.lean [hunk 2]
```diff
@@ -2200,20 +2216,96 @@ theorem union_sdiff_self (s t : Finset α) : (s ∪ t) \ t = s \ t :=
...
 theorem sdiff_singleton_eq_erase (a : α) (s : Finset α) : s \ singleton a = erase s a := by
   ext
   rw [mem_erase, mem_sdiff, mem_singleton, and_comm]
 #align finset.sdiff_singleton_eq_erase Finset.sdiff_singleton_eq_erase
 
+-- This lemma matches `finset.insert_eq` in functionality.
```
Comment (eric-wieser) [inline Mathlib/Data/Finset/Basic.lean:2226]: as above. Perhaps run `fix_comments.py`?

**Review by eric-wieser — APPROVED** [review]
Thanks! I've checked the whole diff and it looks like you got everything. Some of the comments still refer to the mathlib3 names though.

File: Mathlib/Data/Finset/Basic.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Finset/Basic.lean:2226]: I just tried it and it missed "`finset.union_singleton`/`finset.singleton_union`".

**Comment by Vierkantor** [comment]
You are absolutely right about assigned meaning "I will look at this when I have time", in fact I had selected the reply box and was writing `bors r+` myself when your comment appears.

**Pull request closed without merging**